app/com/github/pedrovgs/dotto/core/algebra/dotto.scala

```diff
@@ -0,0 +1,27 @@
+package com.github.pedrovgs.dotto.core.algebra
+
+import cats.free.Free
+import com.github.pedrovgs.dotto.core.types.{LedInteraction, MorseSymbol}
+
+/**
+  * Application algebra used to describe what our program does. Implemented using Cats as core library.
+  */
+object dotto {
+
+  type Message = String
+  type MorseSentence = Seq[Seq[MorseSymbol]]
+  type LedInteractions = Seq[LedInteraction]
+
+  sealed trait Instruction[A]
+
+  case class EnqueueMessage(messageToEnqueue: Message) extends Instruction[Message]
+  case class TranslateMessageIntoMorse(messageToTranslate: Message) extends Instruction[MorseSentence]
+  case class TranslateIntoLedInteractions(morseSentence: MorseSentence) extends  Instruction[LedInteractions]
+  case class ShowLedInteractions(messageToEnqueue: LedInteractions) extends Instruction[LedInteractions]
+
+  def enqueueMessage(messageToEnqueue: Message): Free[Instruction, Message] = Free.liftF(EnqueueMessage(messageToEnqueue))
```

Comment:
I'd group this smart constructors in an object so you can namespace them more easily down the road.